packages/static-build/src/index.ts

```diff
@@ -388,7 +391,20 @@ export const build: BuildV2 = async ({
       }
 
       if (framework.slug === 'gatsby') {
-        await GatsbyUtils.injectPlugins(detectedVersion, entrypointDir);
+        const injectedPlugins = await GatsbyUtils.injectPlugins(
+          detectedVersion,
+          entrypointDir
+        );
+
+        if (injectedPlugins) {
+          const packageManager = await detectFramework({
+            fs: localFileSystemDetector,
+            frameworkList: packageManagers,
+          });
+          if (packageManager === 'pnpm') {
+            await execCommand('pnpm install --lockfile-only');
```

Comment:
I think this is okay but I dont think we use detectFramework when running `pnpm install` in build-utils currently. So it might have slightly different detection logic.